packages/graphile-build-pg/src/plugins/PgColumnsPlugin.ts

```diff
@@ -1,8 +1,31 @@
-// @flow
-import type { Plugin } from "graphile-build";
+import { Plugin, ContextGraphQLObjectTypeFieldsField } from "graphile-build";
+import { ResolveTree } from "graphql-parse-resolve-info";
+import { PgTypeModifier } from "./PgBasicsPlugin";
+import { PgType } from "./PgIntrospectionPlugin";
+import QueryBuilder, { SQL } from "../QueryBuilder";
 
-const nullableIf = (GraphQLNonNull, condition, Type) =>
-  condition ? Type : new GraphQLNonNull(Type);
+type PgGetSelectValueForFieldAndTypeAndModifier = (
+  ReturnType: import("graphql").GraphQLOutputType,
+  fieldContext: ContextGraphQLObjectTypeFieldsField,
+  parsedResolveInfoFragment: ResolveTree,
+  sqlFullName: SQL,
+  type: PgType,
+  typeModifier: PgTypeModifier,
+  parentQueryBuilder: QueryBuilder
+) => SQL;
+
+declare module "graphile-build" {
+  interface Build {
+    pgGetSelectValueForFieldAndTypeAndModifier: PgGetSelectValueForFieldAndTypeAndModifier;
+  }
+}
+
+const nullableIf = <T extends import("graphql").GraphQLNullableType>(
```

Comment:
Why not import these types from "graphql" at the top of this file?

Reply:
It's important that user plugins do not import `graphql` _values_ directly, and instead use the `graphql` available on the `Build` object to help prevent `graphql` version conflicts. To help with this, we're careful to try and avoid `import ... from "graphql"` in the core plugins to lead by example. In `graphile-build` itself this isn't super important, but in `graphile-build-pg` it is.

To help ensure that we only use the _types_ (as opposed to _values_) we use `import` in type positions.

Reply:
Is it possible to make a linting rule for this? I expect `import`s to happen accidentally where the IDE is trying to be helpful and offering automated imports…

Reply:
Yeah; that's exactly what happens. Had to rewrite most of these manually. If you're aware of a lint rule we can enable for this I'm all for it :+1:

Reply:
(In flow you could do `import type` which would only import types and not values. Alas TypeScript does not draw this distinction.)

Reply:
I guess [this one](https://eslint.org/docs/rules/no-restricted-imports) could do it? (Scoped to graphile-build-pg)
```
"no-restricted-imports": ["error", [{
  "name": "graphql",
  "message": "Please refer to build.graphql instead."
}]]
```
Afaik TypeScript only has the feature of not compiling an import if the imported values were only used in type declarations.

Reply:
PR to add that lint rule would be welcome :)